src/pages/Swap.tsx

```diff
@@ -128,11 +127,11 @@ function Swap(): ReactElement {
         to: [],
       }
 
-    const allTokens = Object.values(TOKENS_MAP)
-      .filter(({ isLPToken, addresses }) => !isLPToken && addresses[chainId])
+    const allTokens = Object.values(tokens || {})
+      .filter(({ isLPToken, address }) => !isLPToken && address)
       .filter(({ symbol }) => {
         // get list of pools containing the token
-        const tokenPools = TOKEN_TO_POOLS_MAP[symbol]
+        const tokenPools = tokenToPoolsMap[symbol] ?? []
         // ensure at least one pool is unpaused to include token in swappable list
         const hasAnyUnpaused = tokenPools.some((poolName) => {
           const basicPool = basicPools?.[poolName]
```

Comment:
Going forward, we should do a nullcheck and early exit if any context retrieval is null so we don't have optional chaining scattered everywhere. e.g: 
```
const basicPools = useContext(BasicPoolsContext) or const { tokensMap, tokenToPoolsMap } = useTokenMaps()
```

If we do a nullcheck at the top level for this, we can get rid of this optional on L237
```
tokensMap?.[formStateArg.from.symbol]?.decimals ?? 0,
```

Reply:
Not part of your diff but on L209 of this file, convert ` if (tokenBalances === null || chainId == null)` to use `==` for both comparison for consistency.
And on L213 - L216, use a `[comparison value] == null` so that it checks for both nulls and undefined.

Reply:
> Not part of your diff but on L209 of this file, convert ` if (tokenBalances === null || chainId == null)` to use `==` for both comparison for consistency. And on L213 - L216, use a `[comparison value] == null` so that it checks for both nulls and undefined.

This is ok but there are safer ways to protect against falsy values.

Reply:
`!<comparison value>` which will check it for all falsy values with less syntax.